script/script.js

```diff
@@ -0,0 +1,170 @@
+  // };   ---------
+
+  // console.log(evaluation);
+};
+
+//confirmation user meal type choice
+let mealType = null;
+choice = Number(choice);
+// ...because is string so far. switch uses strict ===
+switch (choice) {
+  case 1:
+    mealType = "Pizza";
+    break;
+  case 2:
+    mealType = "Pasta";
+    break;
+  case 3:
+    mealType = "Salad";
+    break;
+}
+window.alert(`Got it, you picked ${mealType}.`);
+// end step2
+
+// Step 3 - Subtype choice
+const optionsForPizza = `
+1) Margherita
+2) Calzone
+3) Quattro formaggi`;
+const optionsForPasta = `
+1) Bolognese
+2) al olio`;
+const optionsForSalad = `
+1) Salad greek style
+2) Asian rice salad
+3) Tuna salad
+4) Grandmas noodle salad`;
+
+let subTypes = null;
+// holds text options for subtypes of meals
+const mealSubtype = [];
+// gets filled with specific meal options
+let orderSum = 0;
+switch (choice) {
+  case 1:
+    subTypes = optionsForPizza;
+    const pizzaSubtypes = ["Margharita", "Calzone", "Quattro formaggi"];
+    for (let i = 0; i < pizzaSubtypes.length; i++) {
+      mealSubtype.push(pizzaSubtypes[i]);
+    }
+    orderSum = 15;
+    break;
+  case 2:
+    subTypes = optionsForPasta;
+    const pastaSubtypes = ["Bolognese", "al olio"];
+    for (let i = 0; i < pastaSubtypes.length; i++) {
+      mealSubtype.push(pastaSubtypes[i]);
+    }
+    orderSum = 12;
+    break;
+  case 3:
+    subTypes = optionsForSalad;
+    const saladSubtypes = ["Salad greek style", "Asian rice salad", "Tuna salad", "Grandmas noodle salad"];
+    for (let i = 0; i < saladSubtypes.length; i++) {
+      mealSubtype.push(saladSubtypes[i]);
+    }
+    orderSum = 10;
+}
+
+//validation user suboptions mealchoice 
+let subchoice = null;
+// holds a number
+evaluation = false;
+// better reset value
+while (evaluation == false) {
+  let userSubchoice = window.prompt("Please enter the number of disired dish:" + subTypes);
+  if (isNaN(userSubchoice) || userSubchoice <= 0 || userSubchoice > mealSubtype.length) {
+    window.alert(`Sorry, invalid choice, please pick 1 - ${mealSubtype.length} `);
+  } else {
+    evaluation = true;
+  }
+  subchoice = userSubchoice - 1;
+  //otherwise value not visible outside. return not working.
+}
+window.alert(`Good choice! You picked: ${mealSubtype[subchoice]} `);
+// end step3
+
+// Step 4 - Age
+//validation user age
+let age = null;
+evaluation = false;
+while (evaluation == false) {
+  let ageinput = window.prompt(`Please tell us your age, we need to make sure you have legal capacity.`);
+  if (isNaN(ageinput)) {
+    window.alert(`Sorry, please enter a number.`);
+  } else if (ageinput <= 0) {
+    window.alert("Sorry, you are not born yet, please order in 16 years.");
+    throw new Error("Order process terminated.");
+  } else if (ageinput > 0 && ageinput < 16) {
+    window.alert("Sorry, you are too youg for business, please ask your parents to order.")
+    throw new Error("Order process terminated.");
+  } else if (ageinput > 122) {
```

Comment:
Additional validation that I was happy to see when testing the application before reviewing the code :)